modules/light-clients/08-wasm/types/vm.go

```diff
@@ -40,6 +34,26 @@ func (r contractResult) Error() string {
 	return r.ErrorMsg
 }
 
+type statusQueryResponse struct {
+	contractResult
+	Status exported.Status `json:"status"`
+}
+
+type metadataQueryResponse struct {
+	contractResult
+	GenesisMetadata []clienttypes.GenesisMetadata `json:"genesis_metadata"`
+}
+
+type timestampAtHeightQueryResponse struct {
+	contractResult
+	Timestamp uint64 `json:"timestamp"`
+}
+
+type CheckForMisbehaviourExecuteResult struct {
```

Comment:
can be unexported from what I can tell

```suggestion
type checkForMisbehaviourExecuteResult struct {
```